tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3877,3 +3887,849 @@ def test_complex128():
     assert (f_complex64() == test_complex64())
     assert (f_complex128() == test_complex128())
 
+def test_numpy_int(language):
+
+    @types('bool')
+    def test_bool_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('int')
+    def test_int_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('int8')
+    def test_int8_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('int16')
+    def test_int16_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('int32')
+    def test_int32_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('int64')
+    def test_int64_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('float')
+    def test_float_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('float32')
+    def test_float32_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    @types('float64')
+    def test_float64_int(a):
+        import numpy as np
+        b = np.int(a)
+        return b
+
+    import numpy as np
+
+    integer8 = randint(np.iinfo(np.int8(1)).min, np.iinfo(np.int8(1)).max, size=(5), dtype=np.int8)
+    integer16 = randint(np.iinfo(np.int16(1)).min, np.iinfo(np.int16(1)).max, size=(5), dtype=np.int16)
+    integer = randint(min_int, max_int, size=(5), dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=(5), dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=(5), dtype=np.int64)
```

Comment:
Why are you creating arrays here?

Reply:
because we need to pass an array filled with random values with the right data type to imag function, if i did it inside the functions `iinfo` `finfo` and `uniform` will not be recognized by epyccel.

Reply:
Wrong PR? This PR is not about the `imag` function

These arguments are passed to functions such as:
```
@types('int8')
def test_int8_int(a):
    import numpy as np
    b = np.int(a)
    return b
```
which take scalar arguments